### ISSUE TITLE
[BUG] Punch Card Quiz (completePunchCard) Error (solution found)

### ISSUE THREAD

**Issue by AlexyGalaxy** [charlesbel/Microsoft-Rewards-Farmer#147]
I am experiencing an error with the new punch card quiz. This is not a part of the daily set. The issue occurs during completePunchCard.

Script clicks punch card, clicks day 1 quiz, clicks "start quiz”, then script quits out. 
I thnk there is an issue with numberOfQuestions and or counter.

```
counter = str(browser.find_element(By.XPATH, '//*[@id="QuestionPane0"]/div[2]').get_attribute('innerHTML'))[:-1][1:]
numberOfQuestions = max([int(s) for s in counter.split() if s.isdigit()])
```

I am performing tests now if anyone has a solution please post it here.

**Comment by AlexyGalaxy** [comment]
I made a rough fix. This quiz has filled/unfilled bubbles to determin the current question and total questions. I couldnt figure out how to grab that info. If anyone does know please comment below so I can update this code.

Paste the following code under line 539

```
try:
      browser.find_element(By.XPATH, '//*[@id="rqStartQuiz"]').click()
      time.sleep(2)
      questionCounter=1
      totalQuestions=10
      i=0
      for questionCounter in range(totalQuestions) :
          browser.find_element(By.XPATH,'//*[@id="rqAnswerOption' + str(i) + '"]').click()
          time.sleep(2)
          questionAnswer=str(browser.find_element(By.XPATH, '//*[@id="rqAnsStatus"]').get_attribute('innerHTML'))
          print('questionAnswer ='+str(questionAnswer))#delete me testprint
          while questionAnswer.startswith('Oops, try again!') :
              for i  in range (3) :
                  i=i+1
                  browser.find_element(By.XPATH,'//*[@id="rqAnswerOption' + str(i) + '"]').click()
                  time.sleep(2)	
          else :
              pass
          questionCounter = questionCounter + 1
except:
         pass
```

**Comment by AlexyGalaxy** [comment]
This might fix the issue but I would not call this a good and final solution.

```
browser.find_element(By.XPATH, '//*[@id="rqStartQuiz"]').click()
time.sleep(2)
 for x in range (10):
      try :
          time.sleep(4)
          browser.find_element(By.XPATH,'//*[@id="rqAnswerOption0"]').click()
          time.sleep(4)
          browser.find_element(By.XPATH,'//*[@id="rqAnswerOption1"]').click()
          time.sleep(4)
          browser.find_element(By.XPATH,'//*[@id="rqAnswerOption2"]').click()
          time.sleep(4)
          browser.find_element(By.XPATH,'//*[@id="rqAnswerOption3"]').click()
          time.sleep(6)
      except :
          pass
```

**Comment by kingnab** [comment]
i keep getting errors like 
[ERROR] An Error has Occured While Trying to Complete More Promotion Search.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Complete More Promotion Quiz.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Complete More Promotion Quiz.
←[00m
←[91m
[ERROR] An Error has Occured While Trying to Write Err.
←[00m

**Comment by kingnab** [comment]
im also using ur files @AlexyGalaxy   pls help

**Comment by AlexyGalaxy** [comment]
Disable headless mode and watch the script run. There might be some promos that are not working and you might just need to click on them to get the points. If they are actually a quiz the script should be able to solve them. 

As for the ErrorLogger You need to make sure to set that up properly by deleting the # in front of the rewardsErr variable and change the YourNameHere to the name of your computer.  Go to C:\Users to find out the name of your computer

**Comment by AlexyGalaxy** [comment]
I think this is due to the new puzzle pannel. I am not sure how to program the script to complete it. I am working on the script skipping this pannel. When I figure it out I will update my github branch.

**Comment by lollogiaco** [comment]
I am on windows and the path of the rewardsErr works with "\" but when i change ti the code wont work. also if i leave "//" it gives the "rewardsErr not set up correctely"

**Comment by AlexyGalaxy** [comment]
I dont understand please take a picture of your code and try to explain the issue

**Comment by lollogiaco** [comment]
![image](https://user-images.githubusercontent.com/126954743/223197274-0aee59c6-2e43-4f66-87c3-97f892d0e16b.png)
![image](https://user-images.githubusercontent.com/126954743/223197419-30e52869-867f-448a-b552-b07fc8dd3379.png)

**Comment by lollogiaco** [comment]
then i dont undestand why the colors of the texts are not showing up but it's a minor problem

**Comment by kingnab** [comment]
i actually like how you put so much effort into updating the grinder script @AlexyGalaxy

**Comment by lollogiaco** [comment]
still doesn't work

**Comment by AlexyGalaxy** [comment]
That is the line of code that works for me.
```
rewardsErr = 'C://Users//Alex//Desktop//Microsoft-Rewards-Farmer-master//Microsoft.Rewards.Err.txt'
```
Have you tried to set it up to your desktop
```
ewardsErr = 'C://Users//loren//Desktop'
```

**Comment by lollogiaco** [comment]
I put the destination in the GitHub dir and now it worked
also i have a issue that the colors when it's executing doesn't show up, could be a pyton version problem? i have the latest ver 3.10.10 on win 10

**Comment by AlexyGalaxy** [comment]
You only see colors if you use Visual Stuido code. In python cmd you will see black and white

**Comment by lollogiaco** [comment]
i am referring tho when you double click on it not when you edit it

**Comment by AlexyGalaxy** [comment]
right click run as select vsc. once in in vsc hit the play arrow at the top right

**Comment by lollogiaco** [comment]
gets the "Import requests" error

**Comment by AlexyGalaxy** [comment]
no idea what that is. use google to troubleshoot

**Comment by kingnab** [comment]
@AlexyGalaxy  for some reason im getting errors like Error] Desktop Seaches did not Complete.
←[00m
←[93m[INFO] There are 0 Searches Remaining !←[00m
←[93m[INFO] Re-Trying in 259seconds !←[00m

**Comment by kingnab** [comment]
never mind

**Comment by kingnab** [comment]
tank god it now says ←[92m[BING] Finished Desktop and Edge Bing searches 1/3 ! ←[00

**Comment by AlexyGalaxy** [comment]
Theres a more up todate verison out without a reTry function. Its not needed.

**Comment by kingnab** [comment]
how many accounts do you recommend i run?

**Comment by AlexyGalaxy** [comment]
Up to you. The only issue is when you create accounts you cannot use the same phone number more than 5 times bc then the 5th and 6th account get banned. I have 12 accounts running fine. Takes about 9 hours to complete.

**Comment by charlesbel** [comment]
Fixed with v3.0 #188
